src/stdlib/Conversions.scillib

```diff
@@ -0,0 +1,129 @@
+(* ******************************************************************************** *)
+(*   This file is part of scilla.                                                   *)
+(*                                                                                  *)
+(*   Copyright (c) 2018 - present Zilliqa Research Pvt. Ltd.                        *)
+(*                                                                                  *)
+(*   scilla is free software: you can redistribute it and/or modify it under the    *)
+(*   terms of the GNU General Public License as published by the Free Software      *)
+(*   Foundation, either version 3 of the License, or (at your option) any later     *)
+(*   version.                                                                       *)
+(*                                                                                  *)
+(*   scilla is distributed in the hope that it will be useful, but WITHOUT ANY      *)
+(*   WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR  *)
+(*   A PARTICULAR PURPOSE.  See the GNU General Public License for more details.    *)
+(*                                                                                  *)
+(*   You should have received a copy of the GNU General Public License along with   *)
+(*   scilla.  If not, see <http://www.gnu.org/licenses/>.                           *)
+(* ******************************************************************************** *)
+scilla_version 0
+
+import IntUtils
+
+library Conversions
+
+(* Is the ByStr long enough for substr, starting at pos and len. *)
+let is_long_enough : ByStr -> Uint32 -> Uint32 -> Bool =
+  fun (bs : ByStr) =>
+  fun (pos : Uint32) =>
+  fun (len : Uint32) =>
+    let bs_len = builtin strlen bs in
+    let t = builtin add pos len in
+    uint32_le t bs_len
+
+(* Extract out a Scilla type from a ByStr starting at pos. Returns next position. *)
+(* Use the type specific helpers below rather than this function. *)
+let extract_scillaval : forall 'A. forall 'B. (ByStr -> Option 'B) -> ('B -> 'A) -> ByStr -> Uint32 -> Uint32 -> Option (Pair 'A Uint32) =
```

Comment:
Having ... Uint32 -> Uint32 -> ... seems error-prone to me. What if the user swaps the arguments by accident?

Reply:
Do you have a proposed solution for it? We don't have named arguments.

Reply:
Sorry, I missed that it's supposed to be an internal function (I just read the type signatures at first :)) Nevermind.